server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/TasksRoutes.java

```diff
@@ -181,4 +200,24 @@ private TaskId getTaskId(Request req) {
                 .haltError();
         }
     }
+
+    private Duration getTimeout(Request req) {
+        try {
+            String timeout = req.queryParams("timeout");
+            Long value = Optional.ofNullable(timeout)
+                .filter(parameter -> !parameter.isEmpty())
+                .map(Long::parseLong)
+                .orElse(Long.MAX_VALUE / 1000);
+
+            Preconditions.checkState(value > 0);
+            return Duration.ofSeconds(value);
```

Comment:
why do you divide by 1000 and the take a Duration.ofSeconds? You could directly take a Duration.ofMillis, no?

Reply:
ok I got the explanation, but the readability is not very clear. If you name Long.MAX_VALUE as a constant, it would be better.

Reply:
The unit in the API is `seconds` and the maximal value of `Duration` is `Duration.ofMillis(Long.MAX_VALUE)`, I'm forced to divide to avoid overflows.

Reply:
Who will need to wait an HTTP request for 106751991167300 days? What if you assert a max value of 1 day for example?

Reply:
BTW, there's no test to check this "feature"

Reply:
`TasksRoutesTests.getAwaitWithInvalidTimeoutShouldReturnAnError()`

I prefer let the user the choice

Reply:
It doesn't check that you can put 106751991167300 days